### PR TITLE
Create docker-publish.yml (fixes #219)

### LINKED ISSUE #219: Update docker image to 2.6.0
https://hub.docker.com/r/mstruebing/editorconfig-checker/tags shows the latest tag is 2.4.0. Additionally, it lists 2.3.5 on
https://hub.docker.com/r/mstruebing/editorconfig-checker#Quickstart
How can I use version 2.6.0? Can I submit a PR to help implement this bump?

### PR DESCRIPTION
**Pull request by rasa**
Tested and working locally, after setting up the `DOCKER_USERNAME` and `DOCKER_PASSWORD` secrets via Settings > Secrets > Actions.

### CODE REVIEW

**Comment by rasa** [comment]
Fixes #219

**Comment by rasa** [comment]
@mstruebing ping

**Comment by rasa** [comment]
@mstruebing pong. I hope all is well, and you're just busy. No worries if you are.

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# https://github.com/marketplace/actions/docker-metadata-action#basic
+
+name: Publish Docker image
+
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
```
Comment (mstruebing) [inline .github/workflows/docker-publish.yml:13]: I think this needs to be '*.*.*': https://github.com/editorconfig-checker/editorconfig-checker/tags

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# inspired by: 
+# https://docs.github.com/en/actions/publishing-packages/publishing-docker-images and
+# https://github.com/marketplace/actions/docker-metadata-action#basic
+
+name: Publish Docker image
+
+on:
+  push:
+    branches:
+      - 'main'
```
Comment (mstruebing) [inline .github/workflows/docker-publish.yml:11]: Do we really want to run this on `main` everytime?

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (mstruebing) [reply .github/workflows/docker-publish.yml]: Sorry, the automatic markdown formatting kicked in and I did not looked at my comment after I wrote it.

I mean: `*.*.*`

**Review by mstruebing — APPROVED** [review]

**Pull request merged**